src/Chapter3.hs

```diff
@@ -971,6 +1086,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday
+   = Monday
+   | Tuesday
+   | Wednesday
+   | Thursday
+   | Friday
+   | Saturday
+   | Sunday
+   deriving (Show, Eq, Ord, Enum)
+
+isWeekend :: Weekday -> Bool
+-- can this be written down more compact?
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay day = succ day
+
+daysToParty :: Weekday -> Integer
+daysToParty Friday = 0
+daysToParty day = 1 + daysToParty (nextDay day)
```

Comment:
This is a correct implementation, but it is a bit suboptimal. You can use the `fromEnum` method of the `Enum` typeclass and the `mod` function to write this function more efficiently.